### ISSUE TITLE
Experimental Intellisense: Request textDocument/foldingRange failed.

### ISSUE THREAD

**Issue by jasikpark** [mdx-js/vscode-mdx#289]
### Initial checklist

- [X] I read the [support docs](https://mdxjs.com/community/support/)
- [X] I read the [contributing guide](https://mdxjs.com/community/contribute/)
- [X] I agree to follow the [code of conduct](https://github.com/mdx-js/.github/blob/HEAD/code-of-conduct.md)
- [X] I searched [issues](https://github.com/search?q=user%3Amdx-js&type=Issues) and couldn’t find anything (or linked relevant results below)

### Affected packages and versions

v1.2.1 w/ experimental intellisense turned on

### Steps to reproduce

Open a markdown file named `api.mdx`

```mdx
---
title: API Docs
hide_table_of_contents: true
---

import ApiDocMdx from '@theme/ApiDocMdx';

<ApiDocMdx id="api" />
```

### Expected behavior

There should be no false positives / errors when opening a valid file

### Actual behavior

get this error.

```
[Error - 12:14:46 PM] Request textDocument/foldingRange failed.
  Message: Request textDocument/foldingRange failed with message: Cannot read properties of undefined (reading 'textSpan')
  Code: -32603 
```

### Runtime

Node v18

### Package manager

npm v8

### OS

macOS

### Build and bundle tools

Docusaurus

**Comment by remcohaszing** [comment]
Thanks for reporting! I have been able to reproduce the issue.

Does you project have a `tsconfig.json` file? If so, you can add frontmatter support by adding the `mdx` → `plugins` section in it. For example, see our [frontmatter fixture](https://github.com/mdx-js/vscode-mdx/blob/main/fixtures/frontmatter/tsconfig.json).

Configuration for remark plugins is new since version 1.2.0. I welcome any feedback on this.

Regardless of the above, this error should not pop-up. I will look into it soon.

**Comment by jasikpark** [comment]
yep we're using the Typescript starter for Docusaurus in this instance

**Comment by jasikpark** [comment]
i'll test that out 🫡

**Comment by remcohaszing** [comment]
This has just been released in [v1.2.2](https://github.com/mdx-js/vscode-mdx/releases/tag/v1.2.2) :tada:

**Comment by jasikpark** [comment]
wahoo 🔥 excited to try it out
